lib/goldiloader/active_record_patches.rb

```diff
@@ -138,6 +138,29 @@ def eager_loadable?
   end
 end
 
+# In Rails >= 4.1 has_and_belongs_to_many associations create a has_many associations
+# under the covers so we need to make sure to propagate the auto_include option to that
+# association
+if Goldiloader::Compatibility::ACTIVE_RECORD_VERSION >= ::Gem::Version.new('4.1')
+  ActiveRecord::Associations::ClassMethods.class_eval do
+    
+    def has_and_belongs_to_many_with_auto_include_option(name, scope = nil, options = {}, &extension)
+      if scope.is_a?(Hash)
+        options = scope
```

Comment:
No. I need to replicate the logic from https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations.rb#L1789 so I can use the `options` hash.